### ISSUE TITLE
[Talent] Sheath of Light (Retribution Talent) heals 60% of triggering target's HP rather than 60% of spell

### ISSUE THREAD

**Issue by Lantanio** [WoWManiaUK/Redemption#5584]
**What is Happening:**
Upon mind-controlling a Retribution Paladin, Blood Queen can heal up by several million of HP from Sheath of Light. Healing from the talent is many times the magnitude of the spell triggering Sheath of Light in the first place.

**What Should happen:**
Sheath of Light should only heal 60% of the total healing of the ability triggering it (over time). 

Sadly the meter from the ecounter has not been saved. However, the mind-controlled paladin healed the boss by 33k with Flash of Light, which seems appropriate, and the Sheath of Light proc healed for an additional 40 million.  Our best guess is that the talent heals for 60% of the target's total HPin some instances, rather than 60% of the triggering spell.

For reference, 60% of BQL 25hc HP is about 42.8 million. Similar issues may apply to the Lady Death Whisper encounter.

**Comment by Keader** [comment]
It's talent bug, not "icc" one.
Anyway, any tester can try reproduce it?

**Comment by svijetlo** [comment]
must be talent scaling wrong with mind control buff , will see later just got home of work

**Comment by Keader** [comment]
@svijetlo try using .aura 70924 and casting flash light, see if you get high values

**Comment by svijetlo** [comment]
as soon i get to ptr will do

**Comment by svijetlo** [comment]
![image](https://user-images.githubusercontent.com/61908373/128193128-f1127b33-7fe0-4e7d-b3f9-aab5059db5a9.png)
![image](https://user-images.githubusercontent.com/61908373/128193230-8c6d34ae-2ac7-429f-99ee-34cc4e9592fa.png)

**Comment by Keader** [comment]
holy shit hahaha

**Comment by svijetlo** [comment]
btw even under effect of aura icc zone buff effects it could be same while under mind control that icc buff still also efect you what give you even 30% more heals

**Comment by Kilju** [comment]
Not 100% sure what svijetlo meant by his last comment so went to check behavior on live server inside ICC in regards to the 30% healing buff. Not providing exact maths, just approximations so don't take this as theorycraft.

6.3k fol inside ICC (8% extra talents, ~7-7.1k)
1,3k (=5200 total healing) fol ticks ICC (8% extra talents 1500 ticks)
Healed about 80% of the power of fol. With 8% extra healing modifier its closer to 85%
(60*1,3 is 78%)

~5,2k outside ICC fol (with 8% extra healing from 1/5 divinity and Imp Devotion aura = 5,4k)
800 fol ticks. = 3200 total healing (With 8% extra 880 ticks = 3520 total)
Roughly 60% without talents, with talents around 64-65%

Sheath of Light is calculated to my knowledge as resulting Flash of Light heal -> 15% of the healing as 4 ticks -> Any modifiers for each tick. So Divinity or  Druid aura/Improved Devotion aura will push it further outside of ICC.

Do note that critting multiple times will increase the hot part, if you crit 8 times in a row, the tick will be 8x 60% of the fol for the time between first and last crit. After 15s has passed from the first crit, the healing of the hot will go down.

TL;DR Version:

Flash of light gets modified by ICC or MC buff first by 1000/30% and then Sheath of Light gets 60% of Flash of Light healing that gets boosted again for 30%/1000% of which I didnt find quickly anything that this wouldn't be the case as its a new spell entirely. Purge the boss next time

Sheathe stacks as it should with rough mathematics and falls off after 15 seconds from apply. Not sure if svijet took this into consideration for those SS.
